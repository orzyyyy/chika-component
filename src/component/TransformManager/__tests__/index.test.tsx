import React from 'react';
import { mount } from 'enzyme';
import 'nino-cli/scripts/setup';
let TransformManager, Item, targetModule;
switch (process.env.LIB_DIR) {
  case 'lib':
    targetModule = require('../../../../lib/component/TransformManager');
    TransformManager = targetModule.default;
    Item = targetModule.TransformManagerItem;
    break;
  default:
    targetModule = require('..');
    TransformManager = targetModule.default;
    Item = targetModule.TransformManagerItem;
    break;
}

describe('TransformManager', () => {
  it('render correctly', () => {
    const wrapper = mount(
      <TransformManager currentGroup="group1" currentOrder="0">
        <Item group="group1" order={0} key="group1-0">
          group1-0
        </Item>
        <Item group="group1" order={1} key="group1-1">
          group1-1
        </Item>
      </TransformManager>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('transform should work correctly', () => {
    const wrapper = mount(
      <TransformManager currentGroup="group1" currentOrder="0">
        <Item group="group1" order={0} key="group1-0">
          group1-0
        </Item>
        <Item group="group1" order={1} key="group1-1">
          group1-1
        </Item>
      </TransformManager>,
    );
    wrapper.setProps({ currentOrder: '1' });
    expect(
      wrapper
        .find('.Transform-item')
        .at(1)
        .prop('style'),
    ).toEqual({ transform: 'translate3d(0%, 0, 0)' });
  });
});
