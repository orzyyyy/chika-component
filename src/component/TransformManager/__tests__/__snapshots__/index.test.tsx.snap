// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransformManager render correctly 1`] = `
<div
  className="TransformManager"
>
  <div
    className="Transform-item"
    group="group1"
    key="group1-0"
    style={
      Object {
        "transform": "translate3d(0%, 0, 0)",
      }
    }
  >
    group1-0
  </div>
  <div
    className="Transform-item"
    group="group1"
    key="group1-1"
    style={
      Object {
        "transform": "translate3d(100%, 0, 0)",
      }
    }
  >
    group1-1
  </div>
</div>
`;
